src/client-side-encryption/auto_encrypter.ts

```diff
@@ -366,7 +366,12 @@ export class AutoEncrypter {
       return client;
     } catch (error) {
       const { message } = error;
-      if (message && (message.match(/timed out after/) || message.match(/ENOTFOUND/))) {
+      if (
+        message &&
+        (message.match(/timed out after/) ||
+          message.match(/ENOTFOUND/) ||
+          message.match(/ECONNREFUSED/))
```

Comment:
Locally, for the `should return a useful message if mongocryptd fails to autospawn` test, I see errors matching `timed out after` but in CI we get ECONNREFUSED.  I can revert this if we'd like and make the matching in the test a bit more robust, but I thought a better error message might be nice-to-have.

I'm also open to just removing this if-statement entirely.  If we make it into the catch block, we know _something_ went wrong connecting to mongocryptd.  So, the error message seems accurate?  I can't think of a scenario where we wouldn't want to wrap `error` in a more specific error.

Reply:
I think I'm of the preference of removing the if statement entirely and letting the root cause be in the cause field.

Reply:
Yea I think we should unconditionally wrap since this error is really just trying to say "connect threw". _Maybe_ we wouldn't wrap if it is already one of _our_ errors but probably not worth the extra logic.

Reply:
The error should always be a server selection error.  keep in mind that we instantiate this MongoClient - if there were invalid argument errors, we'd be the source of these errors and we'd throw them in our tests.